src/server/routes/api/document/index.js

```diff
@@ -544,10 +544,10 @@ http.post('/', function( req, res, next ){
   };
 
   checkRequestContext( provided )
-    .then( () => res.end() )
     .then( () => createSecret({ secret }) )
     .then( loadTables )
     .then( handleDocCreation )
+    .then( doc => { res.end(); return doc; } )
```

Comment:
Should something like this be at the very end (just before `next`)?  Maybe we should use [`sendStatus()`](http://expressjs.com/en/5x/api.html#res.sendStatus) instead

Reply:
Or maybe the response should be the document JSON

Reply:
I'll return JSON as this is part of my upcoming PR for #675 where the doc link is provided in the success for CTA.

Reply:
Updated to return JSON at end, but you might notice it's a bit laggy (2-3s to end response)

Reply:
Is it the email part creating most of the lag?

Reply:
Yes, that's my best guess just having observed the server console logging

Reply:
Maybe that part could be after the sendStatus().  The main thing is that the document JSON is stable before sending off the http response.

Reply:
Yes I think that's the solution.

Reply:
Sounds good